### DISCUSSION TITLE
the js-live-example doesn't work in safari

### DISCUSSION THREAD

**Discussion by darrellsilver** [deepgram/community#557] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

The "Demo: Transcribe your voice live" page works wonderfully in Chrome / Safari, but the linked repo 
https://github.com/deepgram-devs/js-live-example  only works in Chrome, and NOT Safari.

The issue seems similar to a couple other posts here and many others elsewhere: Safari doesn't support audio/webm. 

But somehow Deepgram's own demo does it properly! So I wish I could just replicate it.

Specifically, in Safari using `js-live-example` the connection closes immediately once the first message is sent.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by darrellsilver** [comment]
Intruigingly, @lukeocodes suggests [here](https://github.com/orgs/deepgram/discussions/222#discussioncomment-6223434) that a polyfill may be necessary for Safari. But the demo code doesn't show that, and the polyfill I've played with doesn't work for Deepgram. 

Ideally the deepgram `js-live-example` code would include this polyfill (or equivalent) such that the world would be a happier place!

**Reply by lukeocodes** [reply]
Yeh this is a good point. Our demos get limited attention at the moment but I have a working Safari example somewhere, I'll try and pop an update on the demo. We have a few microphone demos that probably have the same issue

**Reply by darrellsilver** [reply]
awesome! Any post of that code would be wonderful (as I assume it's less lift than updating / testing the demo)

**Comment by jpvajda** [comment, marked as answer]
👋  I am closing this Discussion as I reported an issue in the project repo. You can track that here:  https://github.com/deepgram-devs/js-live-example/issues/1
